### PR TITLE
MODINVSTOR-1205 Add subjectTypeId and subjectSourceId to instance schema

### PR DESCRIPTION
**Pull request by dmytrokrutii**
### Purpose
Add Subject source and Subject type to schema

### Approach
Extend instance with subjectSourceId and subjectTypeId

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [x] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [x] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MODINVSTOR-1205](https://folio-org.atlassian.net/browse/MODINVSTOR-1205)

### Learning and Resources (if applicable)
_Discuss any research conducted during the development of this pull request. Include links to relevant blog posts, patterns, libraries, or addons that were used to solve the problem._

### Screenshots (if applicable)
Знімок екрана 2024-09-02 о 13 10 19

### CODE REVIEW

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

**Review by RomanChernetskyi — APPROVED** [review]

**Review by KaterynaSenchenko — APPROVED** [review]

**Review by psmagin — COMMENTED** [review]
I am expecting to have some restriction/check that subject type and source ids are exist in appropriate database tables.

File: src/main/resources/templates/db_scripts/subjectIdsReferenceCheckTrigger.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+SELECT subj->>'subjectSourceId'
+INTO invalid_subjectSourceId
+FROM jsonb_array_elements(NEW.jsonb->'subjects') subj
+  LEFT JOIN subject_source ON id = (subj->>'subjectSourceId')::UUID
+WHERE subj->>'subjectSourceId' IS NOT NULL AND id IS NULL
+  LIMIT 1;
+
+IF FOUND THEN
+    RAISE foreign_key_violation USING
+      MESSAGE = 'subjectSourceId doesn''t exist: ' || invalid_subjectSourceId,
```
Comment (psmagin) [inline src/main/resources/templates/db_scripts/subjectIdsReferenceCheckTrigger.sql:17]:
```suggestion
      MESSAGE = 'subject source doesn''t exist: ' || invalid_subjectSourceId,
```

File: src/main/resources/templates/db_scripts/subjectIdsReferenceCheckTrigger.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+SELECT subj->>'subjectTypeId'
+INTO invalid_subjectTypeId
+FROM jsonb_array_elements(NEW.jsonb->'subjects') subj
+  LEFT JOIN subject_type ON id = (subj->>'subjectTypeId')::UUID
+WHERE subj->>'subjectSourceId' IS NOT NULL AND id IS NULL
+  LIMIT 1;
+
+IF FOUND THEN
+    RAISE foreign_key_violation USING
+      MESSAGE = 'subjectTypeId doesn''t exist: ' || invalid_subjectTypeId,
```
Comment (psmagin) [inline src/main/resources/templates/db_scripts/subjectIdsReferenceCheckTrigger.sql:32]:
```suggestion
      MESSAGE = 'subject type doesn''t exist: ' || invalid_subjectTypeId,
```

File: src/main/resources/templates/db_scripts/subjectIdsReferenceCheckTrigger.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                     FOR EACH ROW
+                     WHEN (NEW.jsonb->'subjects' IS NOT NULL AND NEW.jsonb->'subjects' <> '[]')
+                     EXECUTE FUNCTION check_subject_references();
+
+DROP TRIGGER IF EXISTS check_subject_references_on_insert_or_update ON instance CASCADE;
+CREATE TRIGGER check_subject_references_on_insert_or_update
+  BEFORE INSERT OR UPDATE ON instance
+                     FOR EACH ROW
+                     WHEN (NEW.jsonb->'subjects' IS NOT NULL AND NEW.jsonb->'subjects' <> '[]')
+                     EXECUTE FUNCTION check_subject_references();
```
Comment (psmagin) [inline src/main/resources/templates/db_scripts/subjectIdsReferenceCheckTrigger.sql:54]: Why did you define two triggers? Can't it be just `check_subject_references_on_insert_or_update` ?

**Review by psmagin — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Pull request merged**